From: Dean Nelson <dnelson@redhat.com>
Date: Thu, 22 Jul 2010 12:27:51 -0400
Subject: [serial] fix modpost warning in ite887x driver
Message-id: <20100722122751.5885.55192.send-patch@localhost.localdomain>
Patchwork-id: 27026
O-Subject: [RHEL5.6 PATCH 3/4] serial: fix modpost warning in ite887x driver
Bugzilla: 563271
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Resolves RHBZ 563271.

This is triggered if PCI && !HOTPLUG.

  MODPOST vmlinux.o
WARNING: vmlinux.o(.data+0xc910): Section mismatch: reference to .init.text:pci_ite887x_init (between 'pci_serial_quirks' and 'serial_pci_tbl')

Backport of upstream commit:
http://git.kernel.org/linus/f79abb828e1d851387def4247eb3f9477fe03234


diff --git a/drivers/serial/8250_pci.c b/drivers/serial/8250_pci.c
index 98ee5f6..657ac4c 100644
--- a/drivers/serial/8250_pci.c
+++ b/drivers/serial/8250_pci.c
@@ -611,7 +611,7 @@ static int pci_netmos_init(struct pci_dev *dev)
 /* enable IO_Space bit */
 #define ITE_887x_POSIO_ENABLE		(1 << 31)
 
-static int __devinit pci_ite887x_init(struct pci_dev *dev)
+static int pci_ite887x_init(struct pci_dev *dev)
 {
 	/* inta_addr are the configuration addresses of the ITE */
 	static const short inta_addr[] = { 0x2a0, 0x2c0, 0x220, 0x240, 0x1e0,
